### PR TITLE
1245 Student Created Badges

### LINKED ISSUE #1245: Student-created badges
As a student
I should be able to 'propose' badges,
Including setting their names, graphics, and descriptions.

As an instructor
I should be notified that there are student-created badges to review
And be able to edit their elements before approving or rejecting 

The badge should retain info regarding who created it, and if they were created by a student
This should be highlighted in the interface to both students and instructors.

### PR DESCRIPTION
**Pull request by dovek**
### Status
**IN DEVELOPMENT**

### Description
As a student
I should be able to 'propose' badges,
Including setting their names, graphics, and descriptions.

As an instructor
I should be notified that there are student-created badges to review
And be able to edit their elements before approving or rejecting

The badge should retain info regarding who created it, and if they were created by a student
This should be highlighted in the interface to both students and instructors.

### Related PRs
N/A

### Todos
- [ ] Add Tests

### Deploy Notes
N/A

### Gem dependencies
N/A

### Migrations
YES

### Steps to Test or Reproduce
As a student, navigate to the modified 'Badges' page. I will see a new button "Propose Badge". Clicking on the button will take me to the New Badge Form where I can submit a new badge to the instructor of the course.

As an instructor, when I navigate to the 'Badges' page, I will now see 3 tabs: 'Accepted', 'Proposed', and 'Rejected'. By navigating to the show page of one of the proposed badges I will be able to see who submitted the proposed badge. I will also be able to Accept of Reject the badge by using the Options Gear on the show page. Accepting or Rejecting will move it to the appropriate tab on the instructor badges page.

By default, when an instructor creates a badge, it will be set to 'Accepted'

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Badges

======================
Closes #1245

### CODE REVIEW

File: app/views/badges/index.html.haml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 = render partial: "badges/context_menu", locals: {actions: [:new]} if current_user_is_staff?
 
 - if current_user_is_student? || current_user_is_observer?
+  = active_course_link_to decorative_glyph(:plus) + "Propose New #{(term_for :badge).titleize}", new_badge_path, class: "button button-edit"
```
Comment (jonathoy) [inline app/views/badges/index.html.haml]: I noticed that this is generating a list element that's not within a list (ul or ol) and it's causing a weird styling issue.

**Comment by magyarn** [comment]
My commit for adding the `content_name` default param to `active_course_link_to` didn't work as we thought. I was still using the `active_course_link_to_unless_active` function, which gave the impression that it did work. Once I reverted to `active_course_link_to`, the `li` came back. Sorry for the confusion.

**Comment by magyarn** [comment]
@dovek I was able to format the badge award string and keep the localization piece. My styles are ready for review. Thanks!

**Comment by jonathoy** [comment]
@dovek Just trying to clean up some PRs. Do you know what state this work is in? Does it require additional review or is it ready?

**Comment by jonathoy** [comment]
We need a UX resource to go over this feature.

**Pull request closed without merging**